cinema/tests/test_movie_api.py

```diff
@@ -0,0 +1,284 @@
+        )
+        superman_movies = Movie.objects.filter(title="Superman")
+        self.assertEqual(movies.status_code, status.HTTP_403_FORBIDDEN)
+        self.assertEqual(superman_movies.count(), 0)
+
+    def test_get_movie(self):
+        response = self.client.get("/api/cinema/movies/1/")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["title"], "Titanic")
+        self.assertEqual(response.data["description"], "Titanic description")
+        self.assertEqual(response.data["duration"], 123)
+        self.assertEqual(response.data["genres"][0]["name"], "Drama")
+        self.assertEqual(response.data["genres"][1]["name"], "Comedy")
+        self.assertEqual(response.data["actors"][0]["first_name"], "Kate")
+        self.assertEqual(response.data["actors"][0]["last_name"], "Winslet")
+        self.assertEqual(response.data["actors"][0]["full_name"], "Kate Winslet")
+
+    def test_get_invalid_movie(self):
+        response = self.client.get("/api/cinema/movies/100/")
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_put_movie(self):
+        self.client.put(
+            "/api/cinema/movies/1/",
+            {
+                "title": "Watchman",
+                "description": "Watchman description",
+                "duration": 321,
+                "genres": [1, 2],
+                "actors": [1],
+            },
+        )
+        db_movie = Movie.objects.get(id=1)
+        self.assertEqual(
+            [db_movie.title, db_movie.description],
+            [
+                "Titanic",
+                "Titanic description",
+            ],
+        )
+        self.assertEqual(db_movie.title, "Titanic")
+
+    def test_delete_movie(self):
+        response = self.client.delete(
+            "/api/cinema/movies/1/",
+        )
+        db_movies_id_1 = Movie.objects.filter(id=1)
+        self.assertEqual(db_movies_id_1.count(), 1)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+
+class AdminMovieApiTests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.drama = Genre.objects.create(
+            name="Drama",
+        )
+        self.comedy = Genre.objects.create(
+            name="Comedy",
+        )
+        self.actress = Actor.objects.create(first_name="Kate", last_name="Winslet")
+        self.movie = Movie.objects.create(
+            title="Titanic",
+            description="Titanic description",
+            duration=123,
+        )
+        self.movie.genres.add(self.drama)
+        self.movie.genres.add(self.comedy)
+        self.movie.actors.add(self.actress)
+
+        self.user = create_user(
+            username="test_admin",
+            email="test@test.com",
+            password="testpass",
+            is_staff=True,
+        )
+        self.client = APIClient()
+        self.client.force_authenticate(user=self.user)
+
+    def test_get_movies(self):
```

Comment:
You completely don't need to duplicate this check.
It is already implemented in PrivateApiTests.
So it is unnecessary here.